File: csharp/Gherkin.Specs/TokenFormatterBuilder.cs
using System;
using System.Linq;
using System.Text;

namespace Gherkin.Specs
{
	public class TokenFormatterBuilder : IAstBuilder
	{
		private readonly TokenFormatter formatter = new TokenFormatter();
		private readonly StringBuilder tokensTextBuilder = new StringBuilder();

		public string GetTokensText()
		{
			return LineEndingHelper.NormalizeLineEndings(tokensTextBuilder.ToString());
		}

		public void Build(Token token)
		{
			tokensTextBuilder.AppendLine(formatter.FormatToken(token));
		}

		public void StartRule(RuleType ruleType)
		{
			//nop
		}

		public void EndRule(RuleType ruleType)
		{
			//nop
		}

		public object GetResult()
		{
			return new object();
		}
	}
}

File: java/src/main/java/gherkin/TokenFormatterBuilder.java
package gherkin;

public class TokenFormatterBuilder implements Parser.IAstBuilder<String> {
    private final TokenFormatter formatter = new TokenFormatter();
    private final StringBuilder tokensTextBuilder = new StringBuilder();

    @Override
    public void build(Token token) {
        tokensTextBuilder.append(formatter.FormatToken(token)).append("\n");
    }

    @Override
    public void startRule(Parser.RuleType ruleType) {
    }

    @Override
    public void endRule(Parser.RuleType ruleType) {
    }

    @Override
    public String getResult() {
        return tokensTextBuilder.toString();
    }
}
